### PR TITLE
[new release] ppx_tools (6.1+dev+dune)

### LINKED ISSUE #106: ppx_tools port is wrongly versioned and not working on recent compilers
The port of ppx_tools is versioned `7.0+dune` but there's not even a 7.0 version released. It also doesn't work on recent compilers whereas the upstream, released package seems to both build with dune and be compatible with OCaml 4.08 to 4.13.

We need to clean this port. We should look into which version it was built on top of and at the very least change the version here.

### PR DESCRIPTION
**Pull request by NathanReb**
Closes #106 

This is a re-release of the port we currently have labeled as 7.0+dune and that conflicts with the latest releases of ppx_tools which already build with dune.

I'll test this by hand and will also remove the other package as part of this PR.

@Leonidas-from-XIV how can I easily run the linter here?

### CODE REVIEW

**Comment by Leonidas-from-XIV** [comment]
I did it the following way: after adding the repo I used `opam source ppx_tools.6.1+dune+dev` to get it to unpack me the tarball, changed into the folder and ran the linter manually. Somewhat of a manual process but given the CI is supposed to do it it's probably good enough and makes sure it gets the exact archive that opam would pick up and decompresses it using OPAM machinery.

**Comment by NathanReb** [comment]
So to confirm, the linter isn't checking the opam file here but the one in the tarball or upstream URL? Shouldn't it check for both ?

Is the plan to always do it this way or will we eventually consider letting the linter do the pull itself?

**Comment by NathanReb** [comment]
I just ran it and got a single error so far:
```
Version is missing +dune
```

It's a bit hard to tell if it stopped at the first failure or if that's the only one. It might be good to print all the checks and there status (failed or successful) instead.

Also in that specific case we need to find a good convention for what a valid `+dune` version is. I wanted to point out this is somewhere between 6.1 and 6.2. However we choose to represent that, we should make sure the tool supports it.
That being said, it might not be something we wish to support and can make an exception for this old port as well. Writing this I realize it's probably the best solution here.

Any thoughts?

**Comment by Leonidas-from-XIV** [comment]
> So to confirm, the linter isn't checking the opam file here but the one in the tarball or upstream URL? Shouldn't it check for both ?

The linter checks whatever opam files it can find. If we make the CI feed it the opam files from here it will take those, if we make the CI unpack the tarball and run that opam file it will verify those (I made it such because here we just have the `opam` file but in the tarball we also have the `dune-project` which we need to check for `(name)` stanzas). But you're right that we should make it verify whatever it can find, if it is just an `opam` file than just check that. Or have different options on what to check and not to check.

Something like `duniverse-lint --opam-only` could be good since this would check the whole project by default and fails if e.g. `dune-project` is missing and if you just wanted to test the opam file like in this case. What do you think?

**Comment by NathanReb** [comment]
I ran into this again, we need to get this merged, I had even forgotten I opened this PR and re-ran the release

File: packages/ppx_tools/ppx_tools.6.1+dune+dev/opam [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+dev-repo: "git://github.com/dune-universe/ppx_tools.git"
+build: ["dune" "build" "-p" name "-j" jobs
+    "@runtest" {with-test}]
+depends: [
+  "ocaml" {>= "4.08.0" & < "4.12"}
+  "dune" {>= "1.6"}
+]
+url {
+  src:
+    "https://github.com/dune-universe/ppx_tools/releases/download/v6.1%2Bdune%2Bdev/ppx_tools-6.1.dune.dev.tbz"
```
Comment (Leonidas-from-XIV) [inline packages/ppx_tools/ppx_tools.6.1+dune+dev/opam:18]: Wouldn't the right version be `6.1+dev+dune` given it is `6.1+dev` plus a dune port?

**Review by Leonidas-from-XIV — COMMENTED** [review]
I agree that it would be good to avoid having a completely mysterious package version in the repo that would be picked over newer ppx_tools versions with upstream dune support. As such retagging it with the right version is preferable, given 6.4 comes with dune from the get-go.

File: packages/ppx_tools/ppx_tools.6.1+dune+dev/opam (hunk 1 above)
Comment (NathanReb) [reply packages/ppx_tools/ppx_tools.6.1+dune+dev/opam:18]: yep you're right! Would that also make the linter happy?

Comment (NathanReb) [reply packages/ppx_tools/ppx_tools.6.1+dune+dev/opam:18]: Let me fix the whole thing from the start! I'd better have the tag and github releases concur on the version number!

**Review by Leonidas-from-XIV — APPROVED** [review]
:shipit:

File: packages/ppx_tools/ppx_tools.6.1+dune+dev/opam (hunk 1 above)
Comment (NathanReb) [reply packages/ppx_tools/ppx_tools.6.1+dune+dev/opam]: It did fix it!

**Pull request merged**